packages/@uppy/xhr-upload/src/index.js

```diff
@@ -99,21 +99,85 @@ export default class XHRUpload extends BasePlugin {
       /**
        * Check if the response from the upload endpoint indicates that the upload was successful.
        *
-       * @param {number} status the response status code
+       * @param {number} status
+       * @param {string} responseText
        */
-      validateStatus (status) {
+      // eslint-disable-next-line no-unused-vars
+      validateStatus (status, responseText) {
         return status >= 200 && status < 300
       },
     }
 
     this.opts = { ...defaultOptions, ...opts }
     this.i18nInit()
 
-    // Simultaneous upload limiting is shared across all uploads with this plugin.
-    if (internalRateLimitedQueue in this.opts) {
-      this.requests = this.opts[internalRateLimitedQueue]
-    } else {
-      this.requests = new RateLimitedQueue(this.opts.limit)
+    this.uppy.queue.concurrency = this.opts.limit
```

Comment:
it seems odd for a plugin to mutate a deep property on the central uppy instance like this. what if more plugins were to do this, won't it lead to undeterministic behaviour?

Reply:
It's the only way to do backwards compatibility. If we want to do breaking changes for this api as well, that should be discussed

Reply:
ah ok, so before this PR we had one queue per plugin, but now the goal is to have a single queue. I think this has the side-effect that when uploading to multiple destinations, there will be lower performance, because max 6 requests will be enforced in total (as opposed to max 6 requests per destination). But I guess it's an acceptable trade-off in order to get maintainable code. In the future we might want to implement a multi-queue which dispatches requests to the corresponding queue based on which domain name the request will connect to, as I believe that's what the browsers 6 connection limit is based on (per domain)

Reply:
We don't support multiple uploaders, so multiple destinations can only mean to Companion or your own backend for signing, which is only a thing on multipart. I wouldn't overcomplicate the queue in core for one upload plugin. Instead I think it's best we wait till S3 supports HTTP/2.0 and we can have high limits everywhere.